tests/js/JSTest.php

```diff
@@ -88,11 +90,13 @@ public function dataProvider()
         );
 
         // mixture of quotes starting in comment/regex, to make sure strings are
-        // matched correctly, not inside comment/regex.
+        // matched correctly, not inside comment/regex
+        // additionally test catching of empty strings as well
         $tests[] = array(
-            '/abc"def/.test("abc")',
-            '/abc"def/.test("abc")',
+            '/abc"def/.test(\'\')',
```

Comment:
Instead of replacing this test, just add another once. The more test cases, the better :)